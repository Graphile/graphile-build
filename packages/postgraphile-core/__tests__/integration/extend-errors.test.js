const { withPgClient } = require("../helpers");
const { createPostGraphileSchema } = require("../..");

function check(description, sql) {
  test(description, async () => {
    let error;
    await withPgClient(async pgClient => {
      await pgClient.query(sql);
      try {
        await createPostGraphileSchema(pgClient, ["a", "b", "c"], {
          simpleCollections: "both",
        });
      } catch (e) {
        error = e;
      }
    });
    expect(error).toBeTruthy();
    expect(error).toMatchSnapshot();
  });
}

check(
  "simple collections naming clash",
  `
    comment on constraint post_author_id_fkey on a.post is E'@foreignFieldName clash';
  `
);
