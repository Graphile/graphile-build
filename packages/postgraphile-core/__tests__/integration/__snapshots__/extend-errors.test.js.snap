// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple collections naming clash 1`] = `
[Error: A naming conflict has occurred - two entities have tried to define the same key '[1mclash[22m'.

  The first entity was:
  
    [35mBackward relation (connection) for constraint '[1mpost_author_id_fkey[22m' on table [1m"a"."post"[22m. To rename this relation with smart comments:[39m
    [35m[39m
    [35m  COMMENT ON CONSTRAINT "post_author_id_fkey" ON "a"."post" IS E'[1m@foreignFieldName newNameHere[22m\\nRest of existing \\'comment\\' \\nhere.';[39m

  The second entity was:
  
    [33mBackward relation (simple collection) for constraint '[1mpost_author_id_fkey[22m' on table [1m"a"."post"[22m. To rename this relation with smart comments:[39m
    [33m[39m
    [33m  COMMENT ON CONSTRAINT "post_author_id_fkey" ON "a"."post" IS E'[1m@foreignSimpleFieldName newNameHere[22m\\n@foreignFieldName clash\\nRest of existing \\'comment\\' \\nhere.';[39m]
`;

exports[`table naming clash 1`] = `
[Error: A type naming conflict has occurred - two entities have tried to define the same type '[1mPeopleOrderBy[22m'.

  The first entity was:
  
    'addType' call during hook 'PgConnectionArgOrderBy/init/anonymous'

  The second entity was:
  
    'addType' call during hook 'PgConnectionArgOrderBy/init/anonymous']
`;
